tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,318 @@
+import pandas as pd
+import pytest
+import numpy as np
+from string import ascii_lowercase, ascii_uppercase
+from pandas._testing import assert_frame_equal
+
+from janitor.functions import expand_grid
+
+from janitor.utils import _check_instance
+
+
+def test_not_a_dict():
+    """Test that entry(list) is not a dictionary"""
+    data = [60, 70]
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
+
+
+def test_not_a_dict_1():
+    """Test that entry (dataframe) is not a dictionary"""
+    data = pd.DataFrame([60, 70])
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
+
+
+def test_empty_dict():
+    """Test that entry should not be empty"""
+    data = {}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_scalar_to_list():
+    """
+    Test that dictionary values are all converted to lists.
+    """
+    data = {
+        "x": 1,
+        "y": "string",
+        "z": set((2, 3, 4)),
+        "a": tuple((26, 50)),
+        "b": None,
+        "c": 1.2,
+        "d": True,
+        "e": False,
+    }
+    expected = (
+        [],
+        {
+            "x": [1],
+            "y": ["string"],
+            "z": [2, 3, 4],
+            "a": [26, 50],
+            "b": [None],
+            "c": [1.2],
+            "d": [True],
+            "e": [False],
+        },
+    )
+    assert _check_instance(data) == expected
+
+
+def test_nested_dict():
+    """Raise error if dictionary is nested in a dictionary's values"""
+    data = {"x": {"y": 2}}
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
```

Comment:
Assertion not necessary.

```suggestion
        expand_grid(others=data)
```